pandas/core/groupby.py

```diff
@@ -642,9 +642,20 @@ def apply(self, func, *args, **kwargs):
 
         func = self._is_builtin_func(func)
 
-        @wraps(func)
-        def f(g):
-            return func(g, *args, **kwargs)
+        # this is needed so we don't try and wrap strings. If we could
+        # resolve functions to their callable functions prior, this
+        # wouldn't be needed
+        if args or kwargs:
+            if is_callable(func):
+
+                @wraps(func)
+                def f(g):
+                    return func(g, *args, **kwargs)
+            else:
+                raise ValueError('func must be a callable if args or '
+                                 'kwargs are supplied')
+        else:
+            f = func
```

Comment:
> what is a non-callable that actually gets passed here?

`'median'` was getting passed here, once the if / then branching upstream was removed for zero-length indexes. 
Weirdly, the wrapping didn't fail on Python3.

> is there a test where it hits the exception? 

I don't think so, because I don't know anything that passes a string _and_ args. 

What do you think about changing `func = self._is_builtin_func(func)` to convert strings to functions? Is there a function that would do that? Then we could do away with this hack.